libs/platform/auth/src/oauth/oauth.service.spec.ts

```diff
@@ -306,5 +298,44 @@ describe('OauthService', () => {
         mockOauthServiceConfig.loginRoute
       );
     });
+
+    describe('and config specified default provider', () => {
+      const defaultProvider = 'test';
+
+      beforeEach(() => {
+        destroyInjector();
+        createInjector({
+          providers: [
+            ...providers,
+            {
+              provide: OauthServiceConfig,
+              useValue: {
+                ...mockOauthServiceConfig,
+                defaultProvider,
+              },
+            },
+          ],
+        });
+
+        getService().loginWith = vi.fn();
```

Comment:
Please use [`spyOn()` API](https://vitest.dev/guide/mocking.html#functions) for spying on methods instead of overriding them:
```suggestion
        spyOn(getService(), 'loginWith');
```